### DISCUSSION TITLE
script in Custom Queries?

### DISCUSSION THREAD

**Discussion by alpglide** [prest/prest#785] [API Server] [answered]
Custom Queries are really helpful.
It's not always that easy to do everything in one sql.
I know there is plugin.  any  more plugin examples? like how to query the database (for example call preset custom queries inside plugin)?  golang is new to me.

just notice that elements in output of inFormat always come with quote,for example, {{inFormat "name"}} --> ('1','2','3'), not (1,2,3)  while &name=1&name=2&name=3.
Anyway to remove those quote?

**Comment by vmesel** [comment]
Hey @alpglide! They come as quotes because URL fields are strings whenever you pull them.

One way you can use this data is casting it, which is costly, but depending on the table size, it’s a worthy try.

**Comment by alpglide** [comment]
> Hey @alpglide! They come as quotes because URL fields are strings whenever you pull them.
> 
> One way you can use this data is casting it, which is costly, but depending on the table size, it’s a worthy try.

yes, it works. your help is appreciated!

**Comment by arxdsilva** [comment, marked as answer]
> It's not always that easy to do everything in one sql.

@alpglide it is also possible to do multiple queries within the same sql file, just like this:

```sql
WITH pre_query AS ( SELECT * FROM <table> WHERE <conditions>)

SELECT * FROM <table2>
WHERE <table2>.ID=pre_query.ID
AND <conditions2>
```

**Comment by alpglide** [comment]
> 

Yes, that's my preferred way. but other team members do not want to do so. They believe it's hard to understand and maintenance. :)
